starlite/kwargs.py

```diff
@@ -499,26 +508,94 @@ def to_kwargs(self, connection: Union["WebSocket", "Request"]) -> Dict[str, Any]
 
         return output
 
-    async def resolve_dependency(
-        self, dependency: "Dependency", connection: Union["WebSocket", "Request"], **kwargs: Any
-    ) -> Any:
-        """Given an instance of [Dependency][starlite.kwargs.Dependency], recursively resolves its dependency graph.
+    async def resolve_dependencies(self, connection: Union["WebSocket", "Request"], kwargs: Dict[str, Any]) -> None:
+        """Resolve all dependencies into the kwargs, recursively.
 
         Args:
-            dependency: An instance of [Dependency][starlite.kwargs.Dependency]
             connection: An instance of [Request][starlite.connection.Request] or [WebSocket][starlite.connection.WebSocket].
-            **kwargs: Any kwargs to pass recursively.
+            kwargs: Kwargs to pass to dependencies
+        """
+        for batch in self.dependency_batches:
+            await self._resolve_dependencies_concurrently(batch, connection, kwargs)
 
-        Returns:
-            The resolved dependency value
+    async def _resolve_dependencies_concurrently(
+        self,
+        dependencies: Set["Dependency"],
+        connection: Union["WebSocket", "Request"],
+        kwargs: Dict[str, Any],
+    ) -> None:
+        """Resolve the dependencies concurrently. The passed in kwargs will be modified with the results.
+
+        Args:
+            dependencies: An iterable for instances of [Dependency][starlite.kwargs.Dependency].
+            connection: An instance of [Request][starlite.connection.Request] or [WebSocket][starlite.connection.WebSocket].
+            kwargs: Kwargs to pass to dependencies
+        """
+        if not dependencies:
+            pass
+        elif len(dependencies) == 1:
+            await self._resolve_dependency(next(iter(dependencies)), connection, kwargs)
+        else:
+            async with create_task_group() as task_group:
+                for dependency in dependencies:
+                    task_group.start_soon(self._resolve_dependency, dependency, connection, kwargs)
+
+    @staticmethod
+    async def _resolve_dependency(
```

Comment:
this shouldnt be a private static method. Instead it should be the contents of `resolve_dependency` itself, or something else (see my ending comment.